sdk/test-utils/recorder/src/utils.ts

```diff
@@ -52,7 +52,13 @@ export interface RecorderEnvironmentSetup {
 export const env = isBrowser() ? (window as any).__env__ : process.env;
 
 export function isRecordMode() {
-  return env.TEST_MODE === "record";
+  // It should be safe to assume that these two can be considered being in record mode.
+  // For more specific distinctions, one can use isSoftRecordMode.
+  return env.TEST_MODE === "record" || env.TEST_MODE === "soft-record";
+}
+
+export function isSoftRecordMode() {
+  return env.TEST_MODE === "soft-record";
```

Comment:
Can you add some description here?
https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/test-utils/recorder/GUIDELINES.md#setup-for-record-and-playback

Reply:
Done!